tests/support/pytest/ldap.py

```diff
@@ -0,0 +1,392 @@
+                    "-b",
+                    self.base,
+                ],
+            )
+            log.debug("slapd is running")
+            return True
+        except Exception as ex:  # pylint: disable=broad-except
+            log.debug(ex)
+        log.debug("slapd is not running yet")
+        return False
+
+    def __start_check(self, timeout_at) -> bool:
+        while time.time() <= timeout_at:
+            if self.__slapd_running():
+                return True
+            time.sleep(1)
+        return False
+
+    def cmdline(self, *args):
+        cmd = list(super().cmdline(*args))
+        # See the comment in `default_config()` for why the minion runs as root.
+        cmd[2:2] = ["--user", "0:0"]
+        return cmd
+
+    def ldapadd(self, ldif):
+        if isinstance(ldif, str):
+            ldif = ldif.encode()
+        tmpf = tempfile.NamedTemporaryFile(dir=RUNTIME_VARS.TMP, suffix=".ldif")
+        with tmpf as f:
+            f.write(ldif)
+            f.flush()
+            self.__run_unpriv(["cat", f.name])
+            self.__run_unpriv(
+                [
+                    "ldapadd",
+                    "-H",
+                    self.uri,
+                    "-x",
+                    "-D",
+                    self.userdn,
+                    "-w",
+                    self.password,
+                    "-f",
+                    # RUNTIME_VARS.TMP is bind-mounted to RUNTIME_VARS.TMP so
+                    # the path name is the same inside and outside the
+                    # container.
+                    f.name,
+                ],
+            )
+
+    def ldapdelete(self, *dns, recursive=False):
+        cmd = [
+            "ldapdelete",
+            "-H",
+            self.uri,
+            "-x",
+            "-D",
+            self.userdn,
+            "-w",
+            self.password,
+        ]
+        if recursive:
+            cmd.append("-r")
+        cmd.extend(dns)
+        self.__run_unpriv(cmd)
+
+
+@pytest.fixture(scope="module")
+def openldap_minion(salt_master, salt_yaml_cli, salt_key_cli):
+    name = random_string("openldap-minion-")
+    c = salt_master.salt_minion_daemon(
+        name,
+        image="ghcr.io/saltstack/salt-ci-containers/openldap-minion:latest",
+        factory_class=SlapdMinion,
+        pull_before_start=True,
+        skip_on_pull_failure=True,
+        skip_if_docker_client_not_connectable=True,
+    )
+    # For some reason passing this as an argument to the salt_minion_daemon()
+    # call above fails with "TypeError: type object got multiple values for
+    # keyword argument 'python_executable'".
+    c.python_executable = "python3"
+
+    # Workaround for:
+    # https://github.com/saltstack/pytest-salt-factories/issues/139
+    def _after_start():
+        v = importlib_metadata.version("pytest-salt-factories")
+        for cmd in [
+            ("install_packages", "python3-pip"),
+            ("pip", "install", f"pytest-salt-factories=={v}"),
```

Comment:
Salt factories is bind mounted on the minion container

https://github.com/saltstack/pytest-salt-factories/blob/1.0.0rc20/src/saltfactories/daemons/container.py#L772

It's not importable but log forwarding ["should"](https://github.com/saltstack/pytest-salt-factories/blob/1.0.0rc20/src/saltfactories/manager.py#L221-L225) still [work](https://github.com/saltstack/pytest-salt-factories/blob/1.0.0rc20/src/saltfactories/manager.py#L237-L243).